### PR TITLE
various view delta improvements

### PR DESCRIPTION
**Pull request by jacquese**
#### Description
Fixes a bug in expected view delta where sorting didn't happen correctly, and changes the view delta entity change type to be an enum

#### Primary reviewers
@samiwh @jessicafalk

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ComponentTestUtils.h [hunk 1]
```diff
@@ -240,21 +240,26 @@ inline bool CompareEntityComponentCompleteUpdates(const EntityComponentCompleteU
...
 }
 
-inline bool CompareWorkerComponentId(const Worker_ComponentId Lhs, const Worker_ComponentId Rhs)
+inline bool WorkerComponentIdEquality(const Worker_ComponentId Lhs, const Worker_ComponentId Rhs)
 {
 	return Lhs == Rhs;
 }
 
-inline bool CompareWorkerEntityIdKey(const Worker_EntityId Lhs, const Worker_EntityId Rhs)
+inline bool WorkerEntityIdEquality(const Worker_EntityId Lhs, const Worker_EntityId Rhs)
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ComponentTestUtils.h:253]: Possible unnecessery driveby - this is a weird pattern. We could just remove these 3 functions and have an overload to AreEquivalent that doesn't take a predicate.

Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ComponentTestUtils.h:253]: Oh I see (or at least I think I do as I haven't tried to compile it). I think that wasn't done because the functions below are also named AreEquivalent. So That overload can't work. Either those overloads need to be named properly or they should be template specialisations.

Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ComponentTestUtils.h:253]: This can be done in a follow up by whoever though. I just noticed it here.

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp [hunk 2]
```diff
@@ -104,14 +108,34 @@ bool ExpectedViewDelta::Compare(const ViewDelta& Other)
 			return false;
 		}
 
-		if (LhsEntityDelta.bAdded != RhsEntityDelta.bAdded)
+		switch (LhsEntityDelta.Type)
```
Comment (samiwh) [inline SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp:111]: Can this whole thing be replaced by
```
if (LhsEntityDelta.Type != RhsEntityDelta.Type)
{
    return false;
}
```

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp:111]: I think that depends on the ordering of the respective enums in `EntityDelta` and `ExpectedEntityDelta`

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp:111]: Actually can you even compare two different enums?

File: SpatialGDK/Source/SpatialGDK/Public/Tests/SpatialView/ExpectedViewDelta.h [hunk 3]
```diff
@@ -28,7 +29,7 @@ class ExpectedViewDelta
 	ExpectedViewDelta& AddAuthorityLostTemporarily(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId);
 	ExpectedViewDelta& AddDisconnect(const uint8 StatusCode, FString StatusMessage);
 
-	// Compares the Connection Status and the stored Entity Deltas
```
Comment (jessicafalk) [inline SpatialGDK/Source/SpatialGDK/Public/Tests/SpatialView/ExpectedViewDelta.h:31]: oh, it seems like I somehow managed to remove the bits where we tested the connection status before merging it in. That definitely should compare the connection status. we could either do it in this PR or I can add it back in a separate PR

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp (hunk 2 above)
Comment (jessicafalk) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp:111]: could we just use the same enum for both `EntityDelta` and `ExpectedEntityDelta`?

Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp:111]: Oh they're different enums. If the enum isn't anonymous then yes. Otherwise you can give them explicit integer values, but I agree I'd rather not rely on us getting that right.

Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp:111]: If this is the only code this is going to make nicer then I don't care

Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp:111]: this is the only place where we compare these

File: SpatialGDK/Source/SpatialGDK/Public/Tests/SpatialView/ExpectedViewDelta.h (hunk 3 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Public/Tests/SpatialView/ExpectedViewDelta.h:31]: Hmm, I modified this further in https://github.com/spatialos/UnrealGDK/pull/2529 where you can also use it to compare subview deltas. Might be worth doing it in a follow up after that one? I can raise a ticket

File: SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ComponentTestUtils.h (hunk 1 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ComponentTestUtils.h:253]: This was just to fix the bug. If you feel strongly about changing this pattern I can look at it as part of this.

**Review by samiwh — APPROVED** [review]

**Comment by jacquese** [comment]
damn you prow
